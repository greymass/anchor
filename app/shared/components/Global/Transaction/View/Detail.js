// @flow
import React, { Component } from 'react';
import { translate } from 'react-i18next';
import { Label, Table } from 'semantic-ui-react';
import TimeAgo from 'react-timeago';

type Props = {
  transaction: {}
};

class GlobalTransactionViewDetail extends Component<Props> {
  props: Props;
  render() {
    const {
      expired,
      t,
      transaction
    } = this.props;
    const {
      data,
      signed
    } = transaction;
    return (
      <Table definition size="small">
        <Table.Header>
          <Table.Row>
            <Table.HeaderCell>
              {t('property')}
            </Table.HeaderCell>
            <Table.HeaderCell>
              {t('value')}
            </Table.HeaderCell>
          </Table.Row>
        </Table.Header>
        <Table.Body>
          <Table.Row>
            <Table.Cell collapsing>
              {t('global_transaction_view_count')}
            </Table.Cell>
            <Table.Cell>
              1
            </Table.Cell>
          </Table.Row>
          <Table.Row>
            <Table.Cell collapsing>
              {t('global_transaction_view_action_count')}
            </Table.Cell>
            <Table.Cell>
              {data.transaction.transaction.actions.length}
            </Table.Cell>
          </Table.Row>
          <Table.Row>
            <Table.Cell collapsing>
              {t('global_transaction_view_expiration')}
            </Table.Cell>
            <Table.Cell>
              <p>
                <TimeAgo date={`${data.transaction.transaction.expiration}z`} />
                {' '}
                ({data.transaction.transaction.expiration} UTC)
              </p>
              {(expired)
                ? (
                  <React.Fragment>
                    <Label
                      basic
                      color="red"
                      content={t('global_transaction_view_is_expired')}
                    />
                  </React.Fragment>
                )
                : false
              }
            </Table.Cell>
          </Table.Row>
          <Table.Row>
            <Table.Cell collapsing>
              {t('global_transaction_view_status')}
            </Table.Cell>
            <Table.Cell>
              {(signed)
                ? (
                  <Label
                    basic
                    color="green"
                    content={t('global_transaction_view_is_signed') }
                    size="large"
                  />
                )
                : (
                  <Label
                    basic
                    color="orange"
                    content={t('global_transaction_view_is_unsigned')}
                    size="large"
                  />
                )
              }
            </Table.Cell>
          </Table.Row>
        </Table.Body>
      </Table>
    );
  }
}

export default translate('global')(GlobalTransactionViewDetail);
