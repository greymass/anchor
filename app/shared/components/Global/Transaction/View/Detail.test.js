import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { GlobalTransactionViewDetail } from './Detail';

const mockProps = {
  expired: false,
  signed: true,
  t: text => text,
  transaction: {},
};

describe('GlobalTransactionViewDetail', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<GlobalTransactionViewDetail {...mockProps} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

