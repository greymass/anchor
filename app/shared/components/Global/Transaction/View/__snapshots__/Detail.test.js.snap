// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GlobalTransactionViewDetail renders correctly 1`] = `
<Table
  as="table"
  definition={true}
  size="small"
>
  <TableHeader
    as="thead"
  >
    <TableRow
      as="tr"
      cellAs="td"
    >
      <TableHeaderCell
        as="th"
      >
        property
      </TableHeaderCell>
      <TableHeaderCell
        as="th"
      >
        value
      </TableHeaderCell>
    </TableRow>
  </TableHeader>
  <TableBody
    as="tbody"
  >
    <TableRow
      as="tr"
      cellAs="td"
    >
      <TableCell
        as="td"
        collapsing={true}
      >
        global_transaction_view_count
      </TableCell>
      <TableCell
        as="td"
      >
        1
      </TableCell>
    </TableRow>
    <TableRow
      as="tr"
      cellAs="td"
    >
      <TableCell
        as="td"
        collapsing={true}
      >
        global_transaction_view_action_count
      </TableCell>
      <TableCell
        as="td"
      >
        0
      </TableCell>
    </TableRow>
    <TableRow
      as="tr"
      cellAs="td"
    >
      <TableCell
        as="td"
        collapsing={true}
      >
        global_transaction_view_expiration
      </TableCell>
      <TableCell
        as="td"
      >
        <p>
          <TimeAgo
            component="time"
            date="undefinedz"
            formatter={[Function]}
            live={true}
            maxPeriod={Infinity}
            minPeriod={0}
            now={[Function]}
          />
           
          (
           UTC)
        </p>
      </TableCell>
    </TableRow>
    <TableRow
      as="tr"
      cellAs="td"
    >
      <TableCell
        as="td"
        collapsing={true}
      >
        global_transaction_view_status
      </TableCell>
      <TableCell
        as="td"
      >
        <Label
          basic={true}
          color="green"
          content="global_transaction_view_is_signed"
          size="large"
        />
      </TableCell>
    </TableRow>
  </TableBody>
</Table>
`;
